### PR TITLE
Update to rust 1.58

### PR DESCRIPTION
**Pull request by xudong963**
# Which issue does this PR close?

 # Rationale for this change

https://blog.rust-lang.org/2022/01/13/Rust-1.58.0.html

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/src/scalar.rs [hunk 1]
```diff
@@ -1739,7 +1739,7 @@ impl fmt::Display for ScalarValue {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match self {
             ScalarValue::Decimal128(v, p, s) => {
-                write!(f, "{}", format!("{:?},{:?},{:?}", v, p, s))?;
+                write!(f, "{:?},{:?},{:?}", v, p, s)?;
```
Comment (xudong963) [inline datafusion/src/scalar.rs:1742]: https://rust-lang.github.io/rust-clippy/master/index.html#format_in_format_args

**Review by houqp — APPROVED** [review]
LGTM

**Comment by alamb** [comment]
Thanks @xudong963 !

**Comment by xudong963** [comment]
> Thanks @xudong963 !

I wrote this PR without reading the email today, so I didn't notice that you have already filed a ticket. I'm sorry @alamb

**Comment by alamb** [comment]
No worries at all 👍

**Pull request merged**